### PR TITLE
RUMM-2170 Create Feature configuration interfaces

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Create the interfaces for the SDKv2 features configuration
+ renaming to match the RFC

### CODE REVIEW

**Review by mariusc83 — APPROVED** [review]
LGTM

File: dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/FeatureStorageConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.v2.api
+
+/**
+ * Contains the storage configuration for an [FeatureScope] instance.
+ * @property maxItemSize the maximum size (in bytes) for a single item in a batch
+ * @property maxItemPerBatch the maximum number of individual items in a batch
+ * @property maxBatchSize the maximum size of a complete batch
```
Comment (0xnm) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/FeatureStorageConfiguration.kt:13]:
```suggestion
 * @property maxBatchSize the maximum size (in bytes) of a complete batch
```

File: dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/FeatureStorageConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * @property oldBatchThreshold the duration after which a batch is considered too old to be
```
Comment (0xnm) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/FeatureStorageConfiguration.kt:14]: better to specify the unit

File: dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/PayloadFormat.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.v2.api
+
+/**
+ * Describes the format of a paylod for an instance of [SDKFeature].
```
Comment (0xnm) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/v2/api/PayloadFormat.kt:10]:
```suggestion
 * Describes the format of a payload for an instance of [SDKFeature].
```

**Pull request merged**